src/groupNormL2.jl

```diff
@@ -0,0 +1,81 @@
+# Group L2 norm (times a constant)
+
+export GroupNormL2
+
+"""
+**``L_2`` Group - norm**
+    GroupNormL2(λ=1, g = 1, idx = [1,end])
+Returns the function
+```math
+f(x) =  \\sum\\_{i} \\lambda\\_{i}||x\\_{[i]}||\\_2)^{1/2}
+```
+for a nonnegative parameter `λ`.
+  Defaults to NormL2() in ProximalOperators if only 1 group is defined.
+"""
+struct GroupNormL2{
+  R <: Real,
+  RR <: AbstractVector{R},
+  I <: Vector{Vector{Int}}
+  } <: ProximableFunction
+  lambda::RR
+  idx::I
+
+  function GroupNormL2{R, RR,I}(
+    lambda::RR,
+    idx::I
+    ) where {R <: Real, RR <: AbstractVector{R}, I <: Vector{Vector{Int}}}
+    if sum(lambda .< 0) > 1
+      error("weights λ must be nonnegative")
+    elseif length(lambda) != size(idx,1)
+      error("number of weights and indices must be the same")
+    else
+      new{R, RR, I}(lambda, idx)
+    end
+  end
+end
+
+GroupNormL2(
+  lambda::RR = [1.],
+  idx::I = [Int[]],
+) where {R <: Real, RR <: AbstractVector{R}, I <: Vector{Vector{Int}}} =
+GroupNormL2{R, RR, I}(lambda, idx) ## throwing error with idx initialization
+
+function (f::GroupNormL2)(x::AbstractArray{T}) where {T <: Real}
+  sum_c = 0
+  if length(f.idx) == 1
+    return f.lambda * sqrt(sum(x.^2))
+  else
+    for i = 1:length(f.idx)
+      sum_c += f.lambda[i]*sqrt(sum(x[f.idx[i]].^2))
```

Comment:
```suggestion
      sum_c += f.lambda[i] * norm(x[fidx[i]])
```

Reply:
We currently do not use LinearAlgebra as a package, and this would involve adding that. Is that ok?

Reply:
Sure. It's in the standard library, so all users have it.